src/Test/QuickCheck/Monoids.hs

```diff
@@ -0,0 +1,54 @@
+{-# LANGUAGE ExistentialQuantification #-}
+
+module Test.QuickCheck.Monoids
+  ( All (..)
+  , Any (..)
+  ) where
+
+import Data.List.NonEmpty as NonEmpty
+import Data.Semigroup (Semigroup (..))
+import Test.QuickCheck.Property
+
+-- | Conjunction monoid build with `.&&.`.
+--
+-- Use `property @All` as an accessor which doesn't leak
+-- existential variables.
+--
+-- Note: monoid laws are satisfied up to `isSuccess` unless one is using
+-- `checkCoverage`.
+--
+data All = forall p. Testable p => All { getAll :: p }
```

Comment:
Maybe `AllOf` / `AnyOf` or `AllHold` / `AnyHolds` etc.? To unclash with `Data.Monoid`.

Reply:
Or maybe `And` and `Or`.

Reply:
another idea: `Every` / `Some`